### PR TITLE
🗡️ New payments table structure

### PR DESCRIPTION
**Pull request by WRadoslaw**
~Out of scope:~
- [ ] NFT Royalties

### CODE REVIEW

File: packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.tsx [hunk 1]
```diff
@@ -21,54 +33,188 @@ const COLUMNS: TableProps['columns'] = [
...
 }
 
 export const TablePaymentsHistory: FC<TablePaymentsHistoryProps> = ({ data }) => {
+  const [dialogText, setDialogText] = useState('')
   const mappedData: TableProps['data'] = useMemo(
     () =>
       data.map((data) => ({
         date: <Date date={data.date} />,
         type: <Type type={data.type} />,
         amount: <TokenAmount tokenAmount={data.amount} />,
```
Comment (drillprop) [inline packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.tsx:58]: Withdrawal should be negative and it should have red color: 
image
The same applies to revenue share.

File: packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.tsx [hunk 2]
```diff
@@ -21,54 +33,188 @@ const COLUMNS: TableProps['columns'] = [
...
+  sender: string
   amount: BN
+  description?: string
 }
 
 export type TablePaymentsHistoryProps = {
   data: PaymentHistory[]
 }
 
 export const TablePaymentsHistory: FC<TablePaymentsHistoryProps> = ({ data }) => {
```
Comment (drillprop) [inline packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.tsx]: https://www.figma.com/file/ZFRXsyGvnBXnkDUUGc5vyR/My-payments?node-id=2659%3A37306&t=rL7lrMWNVlpYCumY-4 

There is no loading state in the table. Not sure if this was actually implemented.

File: packages/atlas/src/components/Table/Table.styles.ts [hunk 3]
```diff
@@ -41,6 +42,14 @@ export const Th = styled(Text)`
...
   ${cellStyles};
 
+  overflow: hidden;
+
+  p {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
```
Comment (drillprop) [inline packages/atlas/src/components/Table/Table.styles.ts:53]: Probably not your change. But I noticed that cells are slightly bigger than in the design, and it's noticeable(at least for me 😅 ). Could you fix this as well? There is too much padding applied to `td`. 
Design: 
image
Implementation
image

File: packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts [hunk 4]
```diff
@@ -33,3 +35,39 @@ export const StyledJoyTokenIcon = styled(JoyTokenIcon)<{ error: boolean }>`
...
+  background-color: ${cVar('colorBackgroundMutedAlpha')};
+
+  svg {
+    height: 20px;
+
+    g {
+      fill: ${cVar('colorTextMuted')};
+    }
+  }
+`
```
Comment (drillprop) [inline packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts]: Colors don't match the design: 
image
Yours: 
image

**Review by drillprop — CHANGES_REQUESTED** [review]

**Review by drillprop — CHANGES_REQUESTED** [review]
One thing is still incorrect. 
image
The Withdraw widget shouldn't show a negative value.

**Review by drillprop — APPROVED** [review]
Let's wait for final approval from @attemka For me looks pretty neat!

File: packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts (hunk 4 above)
Comment (attemka) [reply packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts]: @WRadoslaw I know its tiny and annoying, but please check the font color as well.
image

**Review by attemka — COMMENTED** [review]
https://github.com/WRadoslaw/atlas/blob/b5c6fa8208668151b3cb4b8626e6a555146b8e70/packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.tsx#L153

please also change blocks -> block

great job overall, only few tiny fixes

File: packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts (hunk 4 above)
Comment (attemka) [reply packages/atlas/src/components/TablePaymentsHistory/TablePaymentsHistory.styles.ts]: @WRadoslaw check this one please sir.

**Pull request merged**